src/Executables/CombineH5/CombineH5.cpp

```diff
@@ -48,62 +54,67 @@ void combine_h5(const std::string& file_prefix, const std::string& subfile_name,
       file_system::glob(file_prefix + "*.h5");
 
   // Checks that volume data was generated with identical versions of SpECTRE
-  if (!check_src_files(file_names)) {
+  if (!h5::check_src_files_match(file_names)) {
     ERROR(
         "One or more of your files were found to have differing src.tar.gz "
         "files, meaning that they may be from differing versions of SpECTRE.");
   }
 
+  // Checks that volume data files contain the same observation ids
+  if (!h5::check_observation_ids_match(file_names, subfile_name)) {
+    ERROR(
+        "One or more of your files were found to have differing observation "
+        "ids, meaning they may be from different runs of your SpECTRE "
+        "executable.");
+  }
+
   // Instantiates the output file and the .vol subfile to be filled with the
   // combined data later
   h5::H5File<h5::AccessType::ReadWrite> new_file(output + "0.h5", true);
```

Comment:
Something I just realized could be problematic: HDF5 has what feels like a thousand different build flags that all make different assumptions. I don't know if it's safe to assume you can have two files open simultaneously. To be safe, I would put braces around these `insert` lines (the file will be closed once the object exits scope), and then do a similar thing in the reading code to ensure you have only one file open. I could see this all falling under "undefined behavior" and so it might work fine sometimes but other times not